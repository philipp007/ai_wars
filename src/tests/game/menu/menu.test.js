import React from 'react';
import Menu from '../../../game/menu/menu';
import MenuItem from '../../../game/menu/menuItem';
import renderer from 'react-test-renderer';
import InputManager from '../../../game/inputManager';
import Keys from '../../../game/inputManager';

test('Menu should be rendered', () => { 
    const menuItems = [
        { id: 1, name: "MenuItem1", selected: true },
        { id: 2, name: "MenuItem2", selected: false },
        { id: 3, name: "MenuItem3", selected: false }
    ]

    let menuRef = null;

    const menu = renderer.create(
        <Menu onRef = { ref => (menuRef = ref) } menuItems = { menuItems } />
    );

    let tree = menu.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Menu should change selected items upon key press', () => { 
    const menuItems = [
        { id: 1, name: "MenuItem1", selected: true },
        { id: 2, name: "MenuItem2", selected: false },
        { id: 3, name: "MenuItem3", selected: false }
    ]

    let menuRef = null;

    const menu = renderer.create(
        <Menu onRef = { ref => (menuRef = ref) } menuItems = { menuItems } />
    );

    let tree = menu.toJSON();
    expect(tree).toMatchSnapshot();

    menuRef.updateItems({ down: true });

    tree = menu.toJSON();
    expect(tree).toMatchSnapshot();

    menuRef.updateItems({ down: true });

    tree = menu.toJSON();
    expect(tree).toMatchSnapshot();
});