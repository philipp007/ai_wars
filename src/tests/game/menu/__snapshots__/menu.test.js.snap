// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should be rendered 1`] = `
<div
  className="centerScreen"
  style={
    Object {
      "fontSize": "80px",
      "top": "25%",
    }
  }
>
  <div
    className="activeMenuItem"
  >
    <span>
      &lt; MenuItem1 &gt;
    </span>
  </div>
  <div
    className="menuItem"
  >
    <span>
      MenuItem2
    </span>
  </div>
  <div
    className="menuItem"
  >
    <span>
      MenuItem3
    </span>
  </div>
</div>
`;

exports[`Menu should change selected items upon key press 1`] = `
<div
  className="centerScreen"
  style={
    Object {
      "fontSize": "80px",
      "top": "25%",
    }
  }
>
  <div
    className="activeMenuItem"
  >
    <span>
      &lt; MenuItem1 &gt;
    </span>
  </div>
  <div
    className="menuItem"
  >
    <span>
      MenuItem2
    </span>
  </div>
  <div
    className="menuItem"
  >
    <span>
      MenuItem3
    </span>
  </div>
</div>
`;

exports[`Menu should change selected items upon key press 2`] = `
<div
  className="centerScreen"
  style={
    Object {
      "fontSize": "80px",
      "top": "25%",
    }
  }
>
  <div
    className="menuItem"
  >
    <span>
      MenuItem1
    </span>
  </div>
  <div
    className="activeMenuItem"
  >
    <span>
      &lt; MenuItem2 &gt;
    </span>
  </div>
  <div
    className="menuItem"
  >
    <span>
      MenuItem3
    </span>
  </div>
</div>
`;

exports[`Menu should change selected items upon key press 3`] = `
<div
  className="centerScreen"
  style={
    Object {
      "fontSize": "80px",
      "top": "25%",
    }
  }
>
  <div
    className="menuItem"
  >
    <span>
      MenuItem1
    </span>
  </div>
  <div
    className="menuItem"
  >
    <span>
      MenuItem2
    </span>
  </div>
  <div
    className="activeMenuItem"
  >
    <span>
      &lt; MenuItem3 &gt;
    </span>
  </div>
</div>
`;
